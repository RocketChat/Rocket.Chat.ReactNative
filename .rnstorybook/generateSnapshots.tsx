import React from 'react';
import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react-native';
import preview from './preview';
import { mockedStore } from '../app/reducers/mockedStore';
import { initStore } from '../app/lib/store/auxStore';

initStore(mockedStore);

export function generateSnapshots(stories: any) {
	describe('Story Snapshots:', () => {
		const composedStories = composeStories(stories, {
			decorators: preview.decorators
		});
		Object.entries(composedStories).forEach(([name, story]) => {
			test(`${name} should match snapshot`, () => {
				const rendered = render(React.createElement(story as React.ComponentType));
				expect(rendered.toJSON()).toMatchSnapshot();
			});
		});
	});
}
