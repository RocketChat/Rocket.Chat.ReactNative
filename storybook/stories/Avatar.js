import React from 'react';
import { ScrollView, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';

import { themes } from '../../app/constants/colors';
import Avatar from '../../app/containers/Avatar/Avatar';
import Status from '../../app/containers/Status/Status';
import StoriesSeparator from './StoriesSeparator';
import sharedStyles from '../../app/views/Styles';

const styles = StyleSheet.create({
	status: {
		borderWidth: 4,
		bottom: -4,
		right: -4
	},
	custom: {
		padding: 16
	}
});

const server = 'https://open.rocket.chat';

const Separator = ({ title, theme }) => <StoriesSeparator title={title} theme={theme} />;
Separator.propTypes = {
	title: PropTypes.string,
	theme: PropTypes.string
};

const AvatarStories = ({ theme }) => (
	<ScrollView style={{ backgroundColor: themes[theme].backgroundColor }}>
		<Separator title='Avatar by text' theme={theme} />
		<Avatar
			text='Avatar'
			server={server}
			size={56}
		/>
		<Separator title='Avatar by roomId' theme={theme} />
		<Avatar
			type='p'
			rid='devWBbYr7inwupPqK'
			server={server}
			size={56}
		/>
		<Separator title='Avatar by url' theme={theme} />
		<Avatar
			avatar='https://user-images.githubusercontent.com/29778115/89444446-14738480-d728-11ea-9412-75fd978d95fb.jpg'
			server={server}
			size={56}
		/>
		<Separator title='Avatar by path' theme={theme} />
		<Avatar
			avatar='/avatar/diego.mello'
			server={server}
			size={56}
		/>
		<Separator title='With ETag' theme={theme} />
		<Avatar
			type='d'
			text='djorkaeff.alexandre'
			avatarETag='5ag8KffJcZj9m5rCv'
			server={server}
			size={56}
		/>
		<Separator title='Without ETag' theme={theme} />
		<Avatar
			type='d'
			text='djorkaeff.alexandre'
			server={server}
			size={56}
		/>
		<Separator title='Emoji' theme={theme} />
		<Avatar
			emoji='troll'
			getCustomEmoji={() => ({ name: 'troll', extension: 'jpg' })}
			server={server}
			size={56}
		/>
		<Separator title='Direct' theme={theme} />
		<Avatar
			text='diego.mello'
			server={server}
			type='d'
			size={56}
		/>
		<Separator title='Channel' theme={theme} />
		<Avatar
			text='general'
			server={server}
			type='c'
			size={56}
		/>
		<Separator title='Touchable' theme={theme} />
		<Avatar
			text='Avatar'
			server={server}
			onPress={() => console.log('Pressed!')}
			size={56}
		/>
		<Separator title='Static' theme={theme} />
		<Avatar
			avatar='https://user-images.githubusercontent.com/29778115/89444446-14738480-d728-11ea-9412-75fd978d95fb.jpg'
			server={server}
			isStatic
			size={56}
		/>
		<Separator title='Custom borderRadius' theme={theme} />
		<Avatar
			text='Avatar'
			server={server}
			borderRadius={28}
			size={56}
		/>
		<Separator title='Children' theme={theme} />
		<Avatar
			text='Avatar'
			server={server}
			size={56}
		>
			<Status
				size={24}
				style={[sharedStyles.status, styles.status]}
				theme={theme}
			/>
		</Avatar>
		<Separator title='Wrong server' theme={theme} />
		<Avatar
			text='Avatar'
			server='https://google.com'
			size={56}
		/>
		<Separator title='Custom style' theme={theme} />
		<Avatar
			text='Avatar'
			server={server}
			size={56}
			style={styles.custom}
		/>
	</ScrollView>
);
AvatarStories.propTypes = {
	theme: PropTypes.string
};
export default AvatarStories;
