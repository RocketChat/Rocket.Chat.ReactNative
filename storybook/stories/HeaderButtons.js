/* eslint-disable import/no-extraneous-dependencies, import/no-unresolved, import/extensions, react/prop-types */
import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { View } from 'react-native';

import * as HeaderButton from '../../app/containers/HeaderButton';
import Header from '../../app/containers/Header';
import { ThemeContext } from '../../app/theme';

const stories = storiesOf('Header Buttons', module);

const HeaderExample = ({ left, right }) => (
	<Header
		headerLeft={left}
		headerTitle={() => <View style={{ flex: 1 }} />}
		headerRight={right}
	/>
);

stories.add('title', () => (
	<>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item title='threads' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item title='threads' />
				</HeaderButton.Container>
			)}
		/>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item title='threads' />
					<HeaderButton.Item title='search' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item title='threads' />
					<HeaderButton.Item title='search' />
				</HeaderButton.Container>
			)}
		/>
	</>
));

stories.add('icons', () => (
	<>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item iconName='threads' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item iconName='threads' />
				</HeaderButton.Container>
			)}
		/>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item iconName='threads' />
					<HeaderButton.Item iconName='search' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item iconName='threads' />
					<HeaderButton.Item iconName='search' />
				</HeaderButton.Container>
			)}
		/>
	</>
));

stories.add('badge', () => (
	<>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item iconName='threads' badge={() => <HeaderButton.Badge tunread={[1]} />} />
					<HeaderButton.Item iconName='threads' badge={() => <HeaderButton.Badge tunread={[1]} tunreadUser={[1]} />} />
					<HeaderButton.Item iconName='threads' badge={() => <HeaderButton.Badge tunread={[1]} tunreadGroup={[1]} />} />
				</HeaderButton.Container>
			)}
		/>
	</>
));

const ThemeStory = ({ theme }) => (
	<ThemeContext.Provider
		value={{ theme }}
	>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item iconName='threads' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item title='Threads' />
					<HeaderButton.Item iconName='threads' badge={() => <HeaderButton.Badge tunread={[1]} />} />
				</HeaderButton.Container>
			)}
		/>
	</ThemeContext.Provider>
);

stories.add('themes', () => (
	<>
		<ThemeStory theme='light' />
		<ThemeStory theme='dark' />
		<ThemeStory theme='black' />
	</>
));

stories.add('common', () => (
	<>
		<HeaderExample
			left={() => (
				<HeaderButton.Drawer />
			)}
		/>
		<HeaderExample
			left={() => (
				<HeaderButton.CloseModal />
			)}
		/>
		<HeaderExample
			left={() => (
				<HeaderButton.CancelModal />
			)}
		/>
		<HeaderExample
			right={() => (
				<HeaderButton.More />
			)}
		/>
		<HeaderExample
			right={() => (
				<HeaderButton.Download />
			)}
		/>
		<HeaderExample
			right={() => (
				<HeaderButton.Preferences />
			)}
		/>
		<HeaderExample
			right={() => (
				<HeaderButton.Legal />
			)}
		/>
	</>
));
