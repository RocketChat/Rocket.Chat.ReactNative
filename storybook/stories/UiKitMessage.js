import React from 'react';
import { ScrollView, StyleSheet, SafeAreaView } from 'react-native';

import { UiKitMessage } from '../../app/containers/UIKit';
import StoriesSeparator from './StoriesSeparator';

// eslint-disable-next-line react/prop-types
const Separator = ({ title }) => <StoriesSeparator title={title} theme='light' />;

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: '#fff'
	},
	padding: {
		paddingHorizontal: 16
	}
});

export default () => (
	<SafeAreaView style={styles.container}>
		<ScrollView style={[styles.container, styles.padding]} keyboardShouldPersistTaps='always'>
			<Separator title='Section' />
			{
				UiKitMessage([{
					type: 'section',
					text: {
						type: 'mrkdwn',
						text: 'Section'
					}
				}])
			}

			<Separator title='Section + Markdown List' />
			{
				UiKitMessage([{
					type: 'section',
					text: {
						type: 'mrkdwn',
						text: '*List*:\n1. Item'
					}
				}])
			}

			<Separator title='Section + Overflow' />
			{
				UiKitMessage([
					{
						type: 'section',
						text: {
							type: 'mrkdwn',
							text: 'Section + Overflow'
						},
						accessory: {
							type: 'overflow',
							options: [
								{
									text: {
										type: 'plain_text',
										text: 'Option 1',
										emoji: true
									},
									value: 'value-0'
								},
								{
									text: {
										type: 'plain_text',
										text: 'Option 2',
										emoji: true
									},
									value: 'value-1'
								},
								{
									text: {
										type: 'plain_text',
										text: 'Option 3',
										emoji: true
									},
									value: 'value-2'
								},
								{
									text: {
										type: 'plain_text',
										text: 'Option 4',
										emoji: true
									},
									value: 'value-3'
								}
							]
						}
					}
				])
			}

			<Separator title='Section + image' />
			{
				UiKitMessage([{
					type: 'section',
					text: {
						type: 'mrkdwn',
						text: 'Section + Image'
					},
					accessory: {
						type: 'image',
						imageUrl: 'https://raw.githubusercontent.com/RocketChat/Rocket.Chat.Artwork/master/Logos/icon-circle-256.png',
						altText: 'plants'
					}
				}])
			}

			<Separator title='Section + button' />
			{
				UiKitMessage([{
					type: 'section',
					text: {
						type: 'mrkdwn',
						text: 'Section + button'
					},
					accessory: {
						type: 'button',
						text: {
							type: 'plain_text',
							text: 'button'
						}
					}
				}])
			}

			<Separator title='Section + Select' />
			{
				UiKitMessage([{
					type: 'section',
					text: {
						type: 'mrkdwn',
						text: 'Section + select'
					},
					accessory: {
						type: 'static_select',
						options: [
							{
								value: 1,
								text: {
									type: 'plain_text',
									text: 'button'
								}
							}, {
								value: 2,
								text: {
									type: 'plain_text',
									text: 'second button'
								}
							}]
					}
				}])
			}

			<Separator title='Section + DatePicker' />
			{
				UiKitMessage([{
					type: 'section',
					text: {
						type: 'mrkdwn',
						text: 'Section + DatePicker'
					},
					accessory: {
						type: 'datepicker',
						initial_date: '1990-04-28',
						placeholder: {
							type: 'plain_text',
							text: 'Select a date',
							emoji: true
						}
					}
				}])
			}

			<Separator title='Section + Multi Select' />
			{
				UiKitMessage([{
					type: 'section',
					text: {
						type: 'mrkdwn',
						text: 'Section + select'
					},
					accessory: {
						type: 'multi_static_select',
						options: [{
							text: {
								type: 'plain_text',
								text: 'button'
							},
							value: 1
						}, {
							text: {
								type: 'plain_text',
								text: 'opt 1'
							},
							value: 2
						}, {
							text: {
								type: 'plain_text',
								text: 'opt 2'
							},
							value: 3
						}, {
							text: {
								type: 'plain_text',
								text: 'opt 3'
							},
							value: 4
						}]
					}
				}])
			}

			<Separator title='Image' />
			{
				UiKitMessage([{
					type: 'image',
					title: {
						type: 'plain_text',
						text: 'Example Image',
						emoji: true
					},
					imageUrl: 'https://raw.githubusercontent.com/RocketChat/Rocket.Chat.Artwork/master/Logos/icon-circle-256.png',
					altText: 'Example Image'
				}])
			}

			<Separator title='Context' />
			{
				UiKitMessage([{
					type: 'context',
					elements: [{
						type: 'image',
						title: {
							type: 'plain_text',
							text: 'Example Image',
							emoji: true
						},
						imageUrl: 'https://raw.githubusercontent.com/RocketChat/Rocket.Chat.Artwork/master/Logos/icon-circle-256.png',
						altText: 'Example Image'
					},
					{
						type: 'mrkdwn',
						text: 'context'
					}
					]
				}])
			}

			<Separator title='Action - Buttons' />
			{
				UiKitMessage([{
					type: 'actions',
					elements: [
						{
							type: 'button',
							text: {
								type: 'plain_text',
								emoji: true,
								text: 'Approve'
							},
							style: 'primary',
							value: 'click_me_123'
						},
						{
							type: 'button',
							text: {
								type: 'plain_text',
								emoji: true,
								text: 'Deny'
							},
							style: 'danger',
							value: 'click_me_123'
						},
						{
							type: 'button',
							text: {
								type: 'plain_text',
								emoji: true,
								text: 'Deny'
							},
							style: 'danger',
							value: 'click_me_123'
						},
						{
							type: 'button',
							text: {
								type: 'plain_text',
								emoji: true,
								text: 'Deny'
							},
							style: 'danger',
							value: 'click_me_123'
						},
						{
							type: 'button',
							text: {
								type: 'plain_text',
								emoji: true,
								text: 'Deny'
							},
							style: 'danger',
							value: 'click_me_123'
						},
						{
							type: 'button',
							text: {
								type: 'plain_text',
								emoji: true,
								text: 'Deny'
							},
							style: 'danger',
							value: 'click_me_123'
						},
						{
							type: 'button',
							text: {
								type: 'plain_text',
								emoji: true,
								text: 'Deny'
							},
							style: 'danger',
							value: 'click_me_123'
						}
					]
				}])
			}

			<Separator title='Fields' />
			{
				UiKitMessage([
					{
						type: 'section',
						fields: [
							{
								type: 'plain_text',
								text: '*this is plain_text text*',
								emoji: true
							},
							{
								type: 'plain_text',
								text: '*this is plain_text text*',
								emoji: true
							},
							{
								type: 'plain_text',
								text: '*this is plain_text text*',
								emoji: true
							},
							{
								type: 'plain_text',
								text: '*this is plain_text text*',
								emoji: true
							},
							{
								type: 'plain_text',
								text: '*this is plain_text text*',
								emoji: true
							}
						]
					}])
			}

			<Separator title='Action - Select' />
			{
				UiKitMessage([{
					type: 'actions',
					elements: [
						{
							type: 'conversations_select',
							placeholder: {
								type: 'plain_text',
								text: 'Select a conversation',
								emoji: true
							}
						},
						{
							type: 'channels_select',
							placeholder: {
								type: 'plain_text',
								text: 'Select a channel',
								emoji: true
							}
						},
						{
							type: 'users_select',
							placeholder: {
								type: 'plain_text',
								text: 'Select a user',
								emoji: true
							}
						},
						{
							type: 'static_select',
							placeholder: {
								type: 'plain_text',
								text: 'Select an item',
								emoji: true
							},
							options: [
								{
									text: {
										type: 'plain_text',
										text: 'Excellent item 1',
										emoji: true
									},
									value: 'value-0'
								},
								{
									text: {
										type: 'plain_text',
										text: 'Fantastic item 2',
										emoji: true
									},
									value: 'value-1'
								},
								{
									text: {
										type: 'plain_text',
										text: 'Nifty item 3',
										emoji: true
									},
									value: 'value-2'
								},
								{
									text: {
										type: 'plain_text',
										text: 'Pretty good item 4',
										emoji: true
									},
									value: 'value-3'
								}
							]
						}
					]
				}])
			}
		</ScrollView>
	</SafeAreaView>
);
