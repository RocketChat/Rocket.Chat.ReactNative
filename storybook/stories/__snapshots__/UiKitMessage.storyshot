// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots UiKitMessage Action - Buttons 1`] = `"{\\"type\\":\\"RCTScrollView\\",\\"props\\":{\\"style\\":[{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"},{\\"paddingHorizontal\\":16}],\\"keyboardShouldPersistTaps\\":\\"always\\"},\\"children\\":[{\\"type\\":\\"View\\",\\"props\\":{},\\"children\\":[{\\"type\\":\\"RCTSafeAreaView\\",\\"props\\":{\\"emulateUnlessSupported\\":true,\\"style\\":{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"}},\\"children\\":null}]}]}"`;

exports[`Storyshots UiKitMessage Action - Select 1`] = `"{\\"type\\":\\"RCTScrollView\\",\\"props\\":{\\"style\\":[{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"},{\\"paddingHorizontal\\":16}],\\"keyboardShouldPersistTaps\\":\\"always\\"},\\"children\\":[{\\"type\\":\\"View\\",\\"props\\":{},\\"children\\":[{\\"type\\":\\"RCTSafeAreaView\\",\\"props\\":{\\"emulateUnlessSupported\\":true,\\"style\\":{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"}},\\"children\\":null}]}]}"`;

exports[`Storyshots UiKitMessage Context 1`] = `"{\\"type\\":\\"RCTScrollView\\",\\"props\\":{\\"style\\":[{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"},{\\"paddingHorizontal\\":16}],\\"keyboardShouldPersistTaps\\":\\"always\\"},\\"children\\":[{\\"type\\":\\"View\\",\\"props\\":{},\\"children\\":[{\\"type\\":\\"RCTSafeAreaView\\",\\"props\\":{\\"emulateUnlessSupported\\":true,\\"style\\":{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"}},\\"children\\":null}]}]}"`;

exports[`Storyshots UiKitMessage Fields 1`] = `"{\\"type\\":\\"RCTScrollView\\",\\"props\\":{\\"style\\":[{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"},{\\"paddingHorizontal\\":16}],\\"keyboardShouldPersistTaps\\":\\"always\\"},\\"children\\":[{\\"type\\":\\"View\\",\\"props\\":{},\\"children\\":[{\\"type\\":\\"RCTSafeAreaView\\",\\"props\\":{\\"emulateUnlessSupported\\":true,\\"style\\":{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"}},\\"children\\":null}]}]}"`;

exports[`Storyshots UiKitMessage Image 1`] = `"{\\"type\\":\\"RCTScrollView\\",\\"props\\":{\\"style\\":[{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"},{\\"paddingHorizontal\\":16}],\\"keyboardShouldPersistTaps\\":\\"always\\"},\\"children\\":[{\\"type\\":\\"View\\",\\"props\\":{},\\"children\\":[{\\"type\\":\\"RCTSafeAreaView\\",\\"props\\":{\\"emulateUnlessSupported\\":true,\\"style\\":{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"}},\\"children\\":null}]}]}"`;

exports[`Storyshots UiKitMessage Section + DatePicker 1`] = `"{\\"type\\":\\"RCTScrollView\\",\\"props\\":{\\"style\\":[{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"},{\\"paddingHorizontal\\":16}],\\"keyboardShouldPersistTaps\\":\\"always\\"},\\"children\\":[{\\"type\\":\\"View\\",\\"props\\":{},\\"children\\":[{\\"type\\":\\"RCTSafeAreaView\\",\\"props\\":{\\"emulateUnlessSupported\\":true,\\"style\\":{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"}},\\"children\\":null}]}]}"`;

exports[`Storyshots UiKitMessage Section + Markdown List 1`] = `"{\\"type\\":\\"RCTScrollView\\",\\"props\\":{\\"style\\":[{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"},{\\"paddingHorizontal\\":16}],\\"keyboardShouldPersistTaps\\":\\"always\\"},\\"children\\":[{\\"type\\":\\"View\\",\\"props\\":{},\\"children\\":[{\\"type\\":\\"RCTSafeAreaView\\",\\"props\\":{\\"emulateUnlessSupported\\":true,\\"style\\":{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"}},\\"children\\":null}]}]}"`;

exports[`Storyshots UiKitMessage Section + Multi Select 1`] = `"{\\"type\\":\\"RCTScrollView\\",\\"props\\":{\\"style\\":[{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"},{\\"paddingHorizontal\\":16}],\\"keyboardShouldPersistTaps\\":\\"always\\"},\\"children\\":[{\\"type\\":\\"View\\",\\"props\\":{},\\"children\\":[{\\"type\\":\\"RCTSafeAreaView\\",\\"props\\":{\\"emulateUnlessSupported\\":true,\\"style\\":{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"}},\\"children\\":null}]}]}"`;

exports[`Storyshots UiKitMessage Section + Overflow 1`] = `"{\\"type\\":\\"RCTScrollView\\",\\"props\\":{\\"style\\":[{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"},{\\"paddingHorizontal\\":16}],\\"keyboardShouldPersistTaps\\":\\"always\\"},\\"children\\":[{\\"type\\":\\"View\\",\\"props\\":{},\\"children\\":[{\\"type\\":\\"RCTSafeAreaView\\",\\"props\\":{\\"emulateUnlessSupported\\":true,\\"style\\":{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"}},\\"children\\":null}]}]}"`;

exports[`Storyshots UiKitMessage Section + Select 1`] = `"{\\"type\\":\\"RCTScrollView\\",\\"props\\":{\\"style\\":[{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"},{\\"paddingHorizontal\\":16}],\\"keyboardShouldPersistTaps\\":\\"always\\"},\\"children\\":[{\\"type\\":\\"View\\",\\"props\\":{},\\"children\\":[{\\"type\\":\\"RCTSafeAreaView\\",\\"props\\":{\\"emulateUnlessSupported\\":true,\\"style\\":{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"}},\\"children\\":null}]}]}"`;

exports[`Storyshots UiKitMessage Section + button 1`] = `"{\\"type\\":\\"RCTScrollView\\",\\"props\\":{\\"style\\":[{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"},{\\"paddingHorizontal\\":16}],\\"keyboardShouldPersistTaps\\":\\"always\\"},\\"children\\":[{\\"type\\":\\"View\\",\\"props\\":{},\\"children\\":[{\\"type\\":\\"RCTSafeAreaView\\",\\"props\\":{\\"emulateUnlessSupported\\":true,\\"style\\":{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"}},\\"children\\":null}]}]}"`;

exports[`Storyshots UiKitMessage Section + image 1`] = `"{\\"type\\":\\"RCTScrollView\\",\\"props\\":{\\"style\\":[{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"},{\\"paddingHorizontal\\":16}],\\"keyboardShouldPersistTaps\\":\\"always\\"},\\"children\\":[{\\"type\\":\\"View\\",\\"props\\":{},\\"children\\":[{\\"type\\":\\"RCTSafeAreaView\\",\\"props\\":{\\"emulateUnlessSupported\\":true,\\"style\\":{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"}},\\"children\\":null}]}]}"`;

exports[`Storyshots UiKitMessage Section 1`] = `"{\\"type\\":\\"RCTScrollView\\",\\"props\\":{\\"style\\":[{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"},{\\"paddingHorizontal\\":16}],\\"keyboardShouldPersistTaps\\":\\"always\\"},\\"children\\":[{\\"type\\":\\"View\\",\\"props\\":{},\\"children\\":[{\\"type\\":\\"RCTSafeAreaView\\",\\"props\\":{\\"emulateUnlessSupported\\":true,\\"style\\":{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"}},\\"children\\":null}]}]}"`;
