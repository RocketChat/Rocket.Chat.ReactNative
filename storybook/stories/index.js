/* eslint-disable import/no-extraneous-dependencies, import/no-unresolved, import/extensions */
import React from 'react';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import { storiesOf } from '@storybook/react-native';

import RoomItem from './RoomItem';
import Message from './Message';
import UiKitMessage from './UiKitMessage';
import UiKitModal from './UiKitModal';
import Markdown from './Markdown';
// import RoomViewHeader from './RoomViewHeader';

import MessageContext from '../../app/containers/message/Context';

// MessageProvider
const baseUrl = 'https://open.rocket.chat';
const user = {
	id: '',
	username: 'diego.mello',
	token: ''
};

// Change here to see themed storybook
const theme = 'light';

const reducers = combineReducers({
	settings: () => ({}),
	login: () => ({
		user: {
			username: 'diego.mello'
		}
	}),
	meteor: () => ({ connected: true }),
	activeUsers: () => ({ abc: { status: 'online', statusText: 'dog' } })
});
const store = createStore(reducers);

const messageDecorator = story => (
	<MessageContext.Provider
		value={{
			user,
			baseUrl,
			onPress: () => {},
			onLongPress: () => {},
			reactionInit: () => {},
			onErrorPress: () => {},
			replyBroadcast: () => {},
			onReactionPress: () => {},
			onDiscussionPress: () => {},
			onReactionLongPress: () => {}
		}}
	>
		{story()}
	</MessageContext.Provider>
);

storiesOf('RoomItem', module)
	.addDecorator(story => <Provider store={store}>{story()}</Provider>)
	.add('list roomitem', () => <RoomItem theme={theme} />);
storiesOf('Message', module)
	.addDecorator(messageDecorator)
	.add('list message', () => <Message theme={theme} />);

storiesOf('UiKitMessage', module)
	.addDecorator(messageDecorator)
	.add('list uikitmessage', () => <UiKitMessage theme={theme} />);
storiesOf('UiKitModal', module)
	.addDecorator(messageDecorator)
	.add('list UiKitModal', () => <UiKitModal theme={theme} />);
storiesOf('Markdown', module)
	.add('list Markdown', () => <Markdown theme={theme} />);

// FIXME: I couldn't make these pass on jest :(
// storiesOf('RoomViewHeader', module)
// 	.add('list', () => <RoomViewHeader theme='black' />);
