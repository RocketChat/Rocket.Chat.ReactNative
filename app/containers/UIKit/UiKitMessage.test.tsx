import { generateSnapshots } from '../../../.rnstorybook/generateSnapshots';
import * as stories from './UiKitMessage.stories';

generateSnapshots(stories);
