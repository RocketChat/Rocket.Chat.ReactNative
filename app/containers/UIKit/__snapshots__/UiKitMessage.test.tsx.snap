// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story Snapshots: ActionButton should match snapshot 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="always"
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      {
        "paddingHorizontal": 16,
      },
    ]
  }
>
  <View />
</RCTScrollView>
`;

exports[`Story Snapshots: ActionSelect should match snapshot 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="always"
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      {
        "paddingHorizontal": 16,
      },
    ]
  }
>
  <View />
</RCTScrollView>
`;

exports[`Story Snapshots: Context should match snapshot 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="always"
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      {
        "paddingHorizontal": 16,
      },
    ]
  }
>
  <View />
</RCTScrollView>
`;

exports[`Story Snapshots: Fields should match snapshot 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="always"
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      {
        "paddingHorizontal": 16,
      },
    ]
  }
>
  <View />
</RCTScrollView>
`;

exports[`Story Snapshots: Image should match snapshot 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="always"
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      {
        "paddingHorizontal": 16,
      },
    ]
  }
>
  <View />
</RCTScrollView>
`;

exports[`Story Snapshots: Section should match snapshot 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="always"
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      {
        "paddingHorizontal": 16,
      },
    ]
  }
>
  <View />
</RCTScrollView>
`;

exports[`Story Snapshots: SectionButton should match snapshot 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="always"
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      {
        "paddingHorizontal": 16,
      },
    ]
  }
>
  <View />
</RCTScrollView>
`;

exports[`Story Snapshots: SectionDatePicker should match snapshot 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="always"
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      {
        "paddingHorizontal": 16,
      },
    ]
  }
>
  <View />
</RCTScrollView>
`;

exports[`Story Snapshots: SectionImage should match snapshot 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="always"
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      {
        "paddingHorizontal": 16,
      },
    ]
  }
>
  <View />
</RCTScrollView>
`;

exports[`Story Snapshots: SectionMarkdownList should match snapshot 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="always"
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      {
        "paddingHorizontal": 16,
      },
    ]
  }
>
  <View />
</RCTScrollView>
`;

exports[`Story Snapshots: SectionMultiSelect should match snapshot 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="always"
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      {
        "paddingHorizontal": 16,
      },
    ]
  }
>
  <View />
</RCTScrollView>
`;

exports[`Story Snapshots: SectionOverflow should match snapshot 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="always"
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      {
        "paddingHorizontal": 16,
      },
    ]
  }
>
  <View />
</RCTScrollView>
`;

exports[`Story Snapshots: SectionSelect should match snapshot 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="always"
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      {
        "paddingHorizontal": 16,
      },
    ]
  }
>
  <View />
</RCTScrollView>
`;
